src/BuildConfig.cc

```diff
@@ -771,9 +772,27 @@ configureBuild(BuildConfig& config, const bool isDebug) {
   if (!fs::exists("src")) {
     throw PoacError("src directory not found");
   }
-  if (!fs::exists("src/main.cc")) {
-    // For now, we only support .cc extension only for the main file.
-    throw PoacError("src/main.cc not found");
+
+  // find main source file
+  fs::path mainSource;
+  for (const auto& entry : fs::directory_iterator("src")) {
+    fs::path path = entry.path();
+    if (!SOURCE_FILE_EXTS.contains(path.extension())) {
+      continue;
+    }
+    if (const auto& s = path.filename().string();
+        s.substr(0, s.find_last_of('.')) != "main") {
```

Comment:
Let's create a small function called `isMainSource` that checks if a string is a main source.